common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -179,12 +179,26 @@ public static StringSearch getStringSearch(
       final UTF8String left,
       final UTF8String right,
       final int collationId) {
+
+    if(collationId == UTF8_BINARY_LCASE_COLLATION_ID) {
+      return getStringSearchUTF8LCase(left, right);
+    }
+
     String pattern = right.toString();
     CharacterIterator target = new StringCharacterIterator(left.toString());
     Collator collator = CollationFactory.fetchCollation(collationId).collator;
     return new StringSearch(pattern, target, (RuleBasedCollator) collator);
   }
 
+  private static StringSearch getStringSearchUTF8LCase(
```

Comment:
I think we can name this: `getStringSearch`
absence of collationId will indicate that no Collator is specified for the corresponding StringSearch instance

Reply:
This is private method and is just being called from `getStringSearch` for `collationId == 1`. It is a question if we should check for UTF8_BINARY_LCASE in the methods calling `getStringSearch` or if we should adapt original `getStringSearch` method to solve this problem.

Reply:
I would prefer:
```
public static StringSearch getStringSearch(final UTF8String pattern, final UTF8String target) {
    return new StringSearch(pattern.toString(), target.toString());
  }
```

and then:
`getStringSearch(left, right)` on line 184

this version of "getStringSearch" may come in handy in the future, so I think we can do this straight away

Reply:
I did it this way now.